### PR TITLE
Feat: Implement Trajectory Thought Pruning in adk-js

### PR DESCRIPTION
**Pull request by AmaadMartin**
Please ensure you have read the contribution guide before creating a pull request.

Link to Issue or Description of Change
1. Link to an existing issue (if applicable):
Closes: #none

2. Or, if no issue exists, describe the change:
This PR implements Trajectory Thought Pruning (ReAct Step Distillation) in `adk-js` to reduce token consumption in long-running agent sessions.

Problem:
Long-running agent sessions consume a large number of tokens because they keep the full history of internal LLM thinking/thought parts from older turns.

Solution:
1. Introduced `TrajectoryThoughtPruningCompactor` that implements `BaseContextCompactor`. It prunes parts with `thought: true` from events older than `eventRetentionSize`.
2. Updated `BaseSessionService` and `DatabaseSessionService` to support upsert in `appendEvent` (updating existing events if the ID matches). This allows persisting pruned events without creating duplicates or causing primary key violations.

Testing Plan
Unit and integration tests have been implemented and pass cleanly.

Unit Tests:
- Added `core/test/context/trajectory_thought_pruning_compactor_test.ts` covering constructor validation, `shouldCompact`, and `compact` logic (including edge cases).
- [x] I have added or updated unit tests for my change.
- [x] All unit tests pass locally.

Integration Tests:
- Added `tests/integration/context_compaction/trajectory_thought_pruning_test.ts` verifying that thoughts are pruned correctly over a multi-turn conversation and that the changes are persisted correctly via `SessionService`.
- Run with: `npx vitest run tests/integration/context_compaction/trajectory_thought_pruning_test.ts`

Checklist
- [x] I have read the CONTRIBUTING.md document.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: core/src/context/trajectory_thought_pruning_compactor.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    );
+
+    return {
+      ...event,
+      content: {
+        ...event.content!,
+        parts: prunedParts,
+      },
+    };
+  }
```
Comment (kalenkevich) [inline core/src/context/trajectory_thought_pruning_compactor.ts]: util functions

**Review by kalenkevich — APPROVED** [review]
